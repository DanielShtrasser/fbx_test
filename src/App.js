import { useState, useMemo } from 'react';
import { withYMaps, YMaps } from '@pbe/react-yandex-maps';

import styles from './App.module.scss';
import MyInput from './components/MyInput/MyInput.jsx';
import PointsList from './components/PointList/PointsList.jsx';
import Map from './components/Map/Map.jsx';


function App() {
  const [points, setPoints] = useState([])
  
  // useEffect(() => {
  //   if(points.length > 1) {
  //     points.map((point, index) => point.priority = index + 1)
  //   }
  // }, [points])

  const ConnectedMyInput = useMemo(() => {
    return withYMaps(MyInput, true, [
      'control.SearchControl',
    ]);
  }, []);

  const ConnectedMap = useMemo(() => {
    return withYMaps(Map, true, [
      'Map',
      'multiRouter.MultiRoute',
      'geocode',
      'control.SearchControl',
      'geoObject.addon.balloon',
      'templateLayoutFactory'
    ]);
  }, []);

  return (
    <div className={styles.app}>
      <YMaps query={{
        lang: 'ru_RU',
        apikey: '3235217d-d276-4978-bf9f-619150230bc6'
        }}>
        <div className={styles.pointsListWrapper}>
          <ConnectedMyInput points={points} setPoints={setPoints} />
          <PointsList points={points} setPoints={setPoints} />
        </div>
        <ConnectedMap points={points} setPoints={setPoints} />
        {/* <Map ymaps={ymaps} points={points} setPoints={setPoints} /> */}
      </YMaps>
    </div>
  );
}

export default App;
