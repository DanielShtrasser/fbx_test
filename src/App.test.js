import React from "react";
import { render } from "@testing-library/react"
import App from './App'

describe("App", () => {
  it("render App", () => {
    const { asFragment } = render(<App />)
    expect(asFragment(<App />)).toMatchSnapshot()
  })
})
