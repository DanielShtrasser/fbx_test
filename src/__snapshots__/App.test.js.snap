// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render App 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div
      class="pointsListWrapper"
    >
      <div />
      <div
        class="pointsList"
      >
        <div />
      </div>
    </div>
    <div />
  </div>
</DocumentFragment>
`;

exports[`App render MyInput 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div
      class="pointsListWrapper"
    >
      <div />
      <div
        class="pointsList"
      >
        <div />
      </div>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
